tests/PSRule.Rules.Azure.Tests/Cmdlet.Common.Tests.ps1

```diff
@@ -531,28 +531,49 @@ InModuleScope -ModuleName 'PSRule.Rules.Azure' {
                 return @(
                     [PSCustomObject]@{
                         Name = 'Resource1'
-                        ResourceID = 'subnetId1'
+                        ResourceID = 'subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.Network/virtualNetworks/vnet-A/subnets/subnet-A'
                     }
                     [PSCustomObject]@{
                         Name = 'Resource2'
-                        ResourceID = 'subnetId2'
+                        ResourceID = 'subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.Network/virtualNetworks/vnet-A/subnets/subnet-B'
                     }
                 )
             };
+
+            Mock -CommandName 'Get-AzResource' -MockWith {
+                return @(
+                    [PSCustomObject]@{
```

Comment:
If not too much work, may lets populate with a basic `microsoft.insights/diagnosticSettings` resource, just update ids and add empty logs array under properties instead of shared key.

Reply:
Definetely not too much work. Was thinking of doing that yesterday. Changed it to a be a diagnostSettings resource instead 🙂